src/shared/bus-util.c

```diff
@@ -508,7 +510,15 @@ int bus_connect_transport_systemd(
                 switch (runtime_scope) {
 
                 case RUNTIME_SCOPE_USER:
-                        return bus_connect_user_systemd(ret_bus);
+                        r = bus_connect_user_systemd(ret_bus);
```

Comment:
Note bus_set_address_user in sd-bus.c still uses DBUS_SESSION_BUS_ADDRESS full address: https://github.com/systemd/systemd/blob/35f51be4f895e04ca513e217c7a0e50156b1ec2b/src/libsystemd/sd-bus/sd-bus.c#L1373

However, this seems to be only used by busctl, so maybe its okay for now since that's an expert tool. Everything else uses APIs in bus-util.c

Reply:
XDG_RUNTIME_DIR works fine but DBUS_SESSION_BUS_ADDRESS fallback still seems to fail...debugging

Reply:
Never mind, user error, it works!
```
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/4711/bus systemctl --user status
```

And invalid DBUS_SESSION_BUS_ADDRESS fails correctly